src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+               (this->timer < 12)) {
+        EnSlime_EngagePlayer(this);
+    } else if (this->timer == 0) {
+        EnSlime_ReturnToIdle(this);
+    }
+}
+
+/**
+ * Initialize reaction to receiving a blunt hit (such as with Goron Punch etc.)
+ * Chuchu does not take damage, but appears slide backwards.
+ * Action function set to UpdateBluntHitReaction.
+ */
+void EnSlime_ReactToBluntHit(EnSlime* this) {
+    this->actor.speed = 10.0f;
+    if (this->actor.velocity.y > 0.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    this->collider.base.acFlags &= ~AC_ON;
+    this->timer = 30;
+    Actor_PlaySfx(&this->actor, NA_SE_EN_SLIME_DEFENCE);
+    if (this->collider.base.ac != NULL) {
+        this->actor.world.rot.y = Actor_WorldYawTowardActor(&this->actor, this->collider.base.ac) + 0x8000;
+    } else {
+        this->actor.world.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+    }
+    this->actionFunc = EnSlime_UpdateBluntHitReaction;
+}
+
+/**
+ * Update reaction to receiving a blunt hit (such as with Goron Punch etc.)
+ * Update XZ speed and jiggle effect
+ * AC_ON flag is off until 15 frames of reaction remain.
+ * When timer is up, chuchu will choose whether to engage player, return home,
+ * or return to idle. If chuchu is too far from home, returning home will
+ * take precedence.
+ */
+void EnSlime_UpdateBluntHitReaction(EnSlime* this, PlayState* play) {
+    f32 timerFactor;
+    f32 scaleFactor;
+
+    EnSlime_UpdateBlink(this);
+    this->timer--;
+    Math_StepToF(&this->actor.speed, 0.0f, 1.0f);
+    timerFactor = sqrtf(this->timer);
+    if (this->timer < 30) {
+        scaleFactor = ((cos_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.08f * timerFactor)) + 1.0f) * 0.01f;
+        this->actor.scale.x = scaleFactor;
+        this->actor.scale.z = scaleFactor;
+    }
+    if (this->timer == 15) {
+        this->collider.base.acFlags |= AC_ON;
+    }
+    this->actor.scale.y = ((sin_rad((f32)this->timer * (2.0f * M_PI / 5.0f)) * (0.07f * timerFactor)) + 1.0f) * 0.01f;
+    this->actor.shape.rot.x = randPlusMinusPoint5Scaled(0x200) * timerFactor;
+    this->actor.shape.rot.z = randPlusMinusPoint5Scaled(0x200) * timerFactor;
+
+    if (this->timer == 0) {
+        if (Actor_WorldDistXZToPoint(&this->actor, &this->actor.home.pos) > this->distLimit) {
+            EnSlime_ReturnHome(this);
+        } else if ((this->actor.xzDistToPlayer < 280.0f) && (Player_GetMask(play) != PLAYER_MASK_STONE)) {
+            EnSlime_EngagePlayer(this);
+        } else {
+            EnSlime_ReturnToIdle(this);
+        }
+    }
+}
+
+/**
+ * Initialize reaction to taking damage.
+ * (I am not confident in what the last arg is for, but it seems to be a boolean of some kind).
+ * Spawn bubbles & splash, start actor rebuff wobble, push actor, and play sfx.
+ * Set action function to UpdateDamageReaction
+ */
+void EnSlime_ReactToDamage(EnSlime* this, PlayState* play, s32 faceDamageSource) {
```

Comment:
Another setup function, we normally call this state "Damaged". So this would be `EnSlime_SetupDamaged`, and the one below would be `EnSlime_Damaged`.